From 369e0b8c62b204cc45fa0b92a9466e590f24c8e2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Wed, 22 Feb 2017 14:59:58 +0000
Subject: [PATCH] don't lock sidebar size due to long master slide names

(cherry picked from commit ea1b27b3b579fd2896c5e054422e909c6739ad86)

and...

let content widgets of sidebar slide bg fill available space

so if its expanded more of the master slide name can be shown

(cherry picked from commit 8fcb253f706d0543e1c2eeb816f74d96cafdc6e2)

Change-Id: I90cbae1cbdea10d5c9bb6678535f73e8419bf8cb
7dcb28e1958edf8ab7fcc03fb77c4b6a9f461ad1
---
 sd/source/ui/sidebar/SlideBackground.cxx          |  3 +++
 sd/uiconfig/simpress/ui/sidebarslidebackground.ui | 16 ++++++++++++----
 2 files changed, 15 insertions(+), 4 deletions(-)

diff --git a/sd/source/ui/sidebar/SlideBackground.cxx b/sd/source/ui/sidebar/SlideBackground.cxx
index dfe65a2..da80c03 100644
--- a/sd/source/ui/sidebar/SlideBackground.cxx
+++ b/sd/source/ui/sidebar/SlideBackground.cxx
@@ -149,6 +149,9 @@ SlideBackground::SlideBackground(
     get(mpPaperSizeBox,"paperformat");
     get(mpPaperOrientation, "orientation");
     get(mpMasterSlide, "masterslide");
+    //let the listbox shrink to any size so the sidebar isn't forced to grow to
+    //the size of the longest master slide name in the document
+    mpMasterSlide->set_width_request(0);
     get(mpFillAttr, "fillattr1");
     get(mpFillGrad, "fillattr2");
     get(mpFillStyle, "fillstyle");
diff --git a/sd/uiconfig/simpress/ui/sidebarslidebackground.ui b/sd/uiconfig/simpress/ui/sidebarslidebackground.ui
index 6e53847..e973eea 100644
--- a/sd/uiconfig/simpress/ui/sidebarslidebackground.ui
+++ b/sd/uiconfig/simpress/ui/sidebarslidebackground.ui
@@ -1,19 +1,20 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!-- Generated with glade 3.18.3 -->
+<!-- Generated with glade 3.20.0 -->
 <interface>
   <requires lib="gtk+" version="3.0"/>
   <requires lib="LibreOffice" version="1.0"/>
   <object class="GtkGrid" id="SlideBackgroundPanel">
     <property name="visible">True</property>
     <property name="can_focus">False</property>
+    <property name="hexpand">True</property>
     <child>
       <object class="GtkGrid" id="grid1">
         <property name="visible">True</property>
         <property name="can_focus">False</property>
-        <property name="halign">center</property>
         <property name="valign">start</property>
         <property name="margin_left">9</property>
         <property name="margin_right">15</property>
+        <property name="hexpand">True</property>
         <property name="border_width">6</property>
         <property name="row_spacing">6</property>
         <property name="column_spacing">6</property>
@@ -46,6 +47,7 @@
           <object class="GtkComboBoxText" id="orientation">
             <property name="visible">True</property>
             <property name="can_focus">False</property>
+            <property name="hexpand">True</property>
             <items>
               <item translatable="yes">Landscape</item>
               <item translatable="yes">Portrait</item>
@@ -60,6 +62,7 @@
           <object class="svxlo-SvxFillTypeBox" id="fillstyle">
             <property name="visible">True</property>
             <property name="can_focus">False</property>
+            <property name="hexpand">True</property>
           </object>
           <packing>
             <property name="left_attach">1</property>
@@ -70,6 +73,7 @@
           <object class="svxlo-ColorLB" id="fillattr">
             <property name="can_focus">False</property>
             <property name="has_tooltip">True</property>
+            <property name="hexpand">True</property>
           </object>
           <packing>
             <property name="left_attach">1</property>
@@ -95,6 +99,7 @@
             <property name="visible">True</property>
             <property name="can_focus">True</property>
             <property name="receives_default">False</property>
+            <property name="hexpand">True</property>
             <property name="xalign">0</property>
             <property name="draw_indicator">True</property>
           </object>
@@ -107,6 +112,7 @@
           <object class="GtkComboBoxText" id="masterslide">
             <property name="visible">True</property>
             <property name="can_focus">False</property>
+            <property name="hexpand">True</property>
           </object>
           <packing>
             <property name="left_attach">1</property>
@@ -118,9 +124,7 @@
             <property name="visible">True</property>
             <property name="can_focus">False</property>
             <property name="halign">end</property>
-            <property name="hexpand">True</property>
             <property name="label" translatable="yes">Master Slide:</property>
-            <property name="ellipsize">end</property>
           </object>
           <packing>
             <property name="left_attach">0</property>
@@ -133,6 +137,7 @@
             <property name="visible">True</property>
             <property name="can_focus">True</property>
             <property name="receives_default">True</property>
+            <property name="hexpand">True</property>
             <property name="action_name">.uno:SelectBackground</property>
           </object>
           <packing>
@@ -144,6 +149,7 @@
           <object class="svxlo-SvxFillAttrBox" id="fillattr1">
             <property name="visible">True</property>
             <property name="can_focus">False</property>
+            <property name="hexpand">True</property>
           </object>
           <packing>
             <property name="left_attach">1</property>
@@ -154,6 +160,7 @@
           <object class="svxlo-ColorLB" id="fillattr2">
             <property name="visible">True</property>
             <property name="can_focus">False</property>
+            <property name="hexpand">True</property>
           </object>
           <packing>
             <property name="left_attach">1</property>
@@ -166,6 +173,7 @@
             <property name="visible">True</property>
             <property name="can_focus">True</property>
             <property name="receives_default">False</property>
+            <property name="hexpand">True</property>
             <property name="xalign">0</property>
             <property name="draw_indicator">True</property>
           </object>
-- 
2.9.3

